pkg/storegateway/indexcache/cache.go

```diff
@@ -116,6 +118,22 @@ type IndexCache interface {
 	FetchLabelValues(ctx context.Context, userID string, blockID ulid.ULID, labelName string, matchersKey LabelMatchersKey) ([]byte, bool)
 }
 
+// BlockTTL determines an appropriate TTL for cache entries related to a particular block
+// based on metadata about the block. Use a shorter TTL for temporary blocks that will be
+// compacted and deleted soon and a longer TTL for larger blocks that won't be deleted.
+func BlockTTL(meta *block.Meta) time.Duration {
```

Comment:
This assumes the default config for:

```
# (advanced) List of compaction time ranges.
# CLI flag: -compactor.block-ranges
[block_ranges: <list of durations> | default = 2h0m0s,12h0m0s,24h0m0s]
```

It's not a blocker, but I'm wondering if we could improve the compactor config `Validate()` to log a warning if there's no range >= 24h configured, saying the system may be inefficient. Then in the code we can comment about this function to remember "why" we put such message.

Reply:
Sure, I'll add a warning in the compactor.